docs/AUTH.md

```diff
@@ -1,36 +1,387 @@
 
-## Access-Matrix Verbs
+## 1. URM Concept
 
-Generally the following verbs are supported:
+URM (User-Rights-Matrix) follows the Map<String, Object> format as below,
+which allows to define a matrix of an **Action** against target **Resource** by specifying one/more **AttributeMaps** of zero/more **Attributes**, that should be compared to validate request authorization:
 
-- **useXXX**: Means that a resource can be used and **id**, **title**, **description** can be viewed.
-- **manageXXX**: Means that a resource can be read and modified.
-- **createXXX**: Allow the creation of a feature. 
-- **readXXX**: Allow reading of a feature.
-- **updateXXX**: Allow to update a feature.
-- **deleteXXX**: Allow to delete a feature.
-- **purgeXXX**: Allow to purge a feature, this is final deletion.
+```json
+{
+    "urm": {
+        "xyz-hub": {
+            // zero or more Actions
+            "<action>": [
+                // one or more Attribute-Map's (atleast one should match)
+                {
+                    // zero or more access Attributes (all should match)
+                    "<accessAttribute1>": "<exactValue>",
+                    "<accessAttribute2>": "<valueWithWildcard*>",
+                    "<accessAttribute3>": [  // one or more values (all should match)
+                        "<anotherExactValue>",
+                        "<anotherValueWithWildcard*>"
+                    ]
+                }
+            ]
+        }
+    }
+}
+```
 
-## Access-Matrix Types
+For example:
 
-- **EventHandler**: 
+```json
+{
+    "urm": {
+        "xyz-hub": {
+            // zero or more Actions
+            "readFeatures": [
+                // one or more Attribute-Map's (atleast one should match)
+                {
+                    // zero or more access Attributes (all should match)
+                    "id": "my-unique-feature-id",
+                    "storageId": "id-with-wild-card-*",
+                    "tags": [  // one or more values (all should match)
+                        "my-unique-tag",
+                        "some-common-tag-with-wild-card-*"
+                    ]
+                }
+            ]
+        }
+    }
+}
+```
+
+
+
+### Action
+
+Defines type of operation allowed against a resource. Refer later sections on this page, to know which specific actions are allowed against which resource.
+
+But, in general:
+
+- **useXXX**: Means that a resource can be used and **id**, **title**, **description** can be viewed, but NOT **properties** details
+- **manageXXX**: Means that a resource can be read and modified (i.e. full access).
+- **createXXX**: Allow the creation of a resource.
+- **readXXX**: Allow reading of a resource.
+- **updateXXX**: Allow to update a resource.
+- **deleteXXX**: Allow to delete a resource.
+
+Each Action can have list of one or more **AttributeMaps**, which is then compared against target Resource AttributeMaps to validate authorization.
```

Comment:
We should document names clearly:
- URM = User-Rights-Matrix holding the rights of the principal.
- ARM = Access-Rights-Matrix holding the rights necessary to perform the **action**.

Technically, all entries in the ARM are compared against all entries in the URM per **action**. For each entry in the ARM at least one entry being greater or equal to the ARM entry must exist to grant access.

Reply:
I am OK with ARM (as convention). Yes, it would be good to add this detail. I will address this.

Reply:
addressed in latest commit